packages/backend-app-api/src/services/implementations/httpAuth/httpAuthServiceFactory.ts

```diff
@@ -232,7 +232,7 @@ class DefaultHttpAuthService implements HttpAuthService {
     const externalBaseUrlStr = await this.#discovery.getExternalBaseUrl(
       this.#pluginId,
     );
-    const externalBaseUrl = new URL(origin ?? externalBaseUrlStr);
+    const externalBaseUrl = new URL(externalBaseUrlStr ?? origin);
```

Comment:
Reading this again, I realise that the `externalBaseUrlStr` will never be empty, so it'll always be picked over the origin value. @Rugvip is it safe you think to actually trust the discovery here, and just ignore origin headers? Or could there be something else at play here

Reply:
As a matter of fact - did this code intend to look at the host header rather than the origin header?

Reply:
Regarding the priority - it should be host first. The external base URL is a fallback just in case, but we prefer using the header in order to support multiple deployments with a single backend. Although happy to be challenged on that, in the auth backend it's very useful to support multiple app origins, but perhaps the value of supporting multiple external URLs is kinda limited?

Reply:
> in the auth backend it's very useful to support multiple app origins, but perhaps the value of supporting multiple external URLs is kinda limited?

mind elaborating on this a bit more so I can understand better?

Reply:
In the auth backend it can be useful to be able to serve multiple different origins, let's say for example you want a single auth backend to serve both your *frontend* staging and production environments those could be hosted on `backstage.acme.org` and `staging.backstage.acme.org`. By reading the origin from the request we're able to generate frame responses that work with both those origins, and hopefully only those. Another perhaps even better example are PR testing environments where each PR gets its own subdomain.

For this case the equivalent would be something like having a single backend that you from the frontend can for example reach both directly at `backstage.internal.acme.org`, but also via some proxy at as `backstage.acme.org`.

And yes, definitely `req.hostname` 👍, we'll want it to work with `X-Forwarded-Host`

Reply:
got it, thanks - updated the change